src/containers/todo_text_input/index.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import * as classnames from 'classnames';
+
+type TodoInputProp = {
+  editing?: boolean;
+  isNew?: boolean;
+  placeholder?: string;
+  value?: string;
+  onBlur?: () => any;
+  onChange?: () => any;
+  onSubmit?: () => any;
+}
+
+export default ({ editing = false, isNew = false, placeholder, value = '', onBlur, onChange, onSubmit }: TodoInputProp) => (
```

Comment:
You can probably replace the last 5 with `...props` and just pass them through, then use `withProps` to set the default value for `value` if you really need it.